Lucid/Core/CoreManager.swift

```diff
@@ -499,9 +499,12 @@ private extension CoreManager {
         return Publishers.QueuedReplayOnce(combineOperationQueue) { promise, completion in
             Task(priority: .high) {
                 do {
-                    completion()
-                    let result = try await self.get(withQuery: query, in: context)
-                    promise(.success(result))
+                    async let result = try await self.get(withQuery: query, in: context)
+                    Task(priority: .low) {
+                        try? await Task.sleep(nanoseconds: NSEC_PER_SEC / 10000) // 0.1 milliseconds
+                        completion()
+                    }
+                    try await promise(.success(result))
```

Comment:
This isn't great, but there is currently a race condition between the `Tasks` operations. Since this is only affecting the Combine code flows, it seemed less risky.

Reply:
How would it work if we call `completion()` after the promise call?

Reply:
The problem there is the completion is now tied to the entire action. The idea should be to process the operations in the order they are received, but not to make the call-stacks actually wait for the previous one to complete.

Reply:
Ahh understood 👍